src/components/tl/ucc_tl.h

```diff
@@ -84,11 +85,31 @@ ucc_status_t ucc_tl_context_get(ucc_context_t *ctx, ucc_tl_type_t type,
                                 ucc_tl_context_t **tl_context);
 ucc_status_t ucc_tl_context_put(ucc_tl_context_t *tl_context);
 
+typedef struct ucc_team_create_multiple_req {
```

Comment:
I would do that little bit differently - all exactly the same logic but looks another way. Tell me what you think.
```
struct team_desc {
    ucc_tl_context_t *ctx;
    ucc_tl_team_t     *team;
    ucc_status_t       status;
    ucc_base_team_params_t p; //value instead of ptr, it is not a critical path and memcpy is not important. but don't 
                                                    need to do extra mallocs
};

struct ucc_team_create_multiple_req {
    int n_teams;
    int last_created;
    team_desc_t descs[1];
};
```
then in you req_alloc
```
req = malloc(sizeof(multiple_req_t) + (n_teams-1)*sizeof(team_desc));
```
and cleanup will be just 1 free.

Everything else will be the same.

Wdyt?

Reply:
looks much better!